### PR TITLE
Improve Illumina 1.5 guess encoding heuristics

### PR DESCRIPTION
**Pull request by cviner**
Adds new heuristics to improve the guessing of Illumina 1.5 encodings.

Resolves issue #18.

### CODE REVIEW

File: reads-utils/guess-encoding.py [hunk 1]
```diff
@@ -46,6 +60,30 @@ def get_encodings_in_range(rmin, rmax, ranges=RANGES):
...
+
+        # 66: Phread+64 quality score of 2 'B'
+        #     used by Illumina 1.5+ as QC indicator
+        elif 66 in map(operator.itemgetter(0),
+                       qual_val_counts.most_common(N_MOST_COMMON_THRESH)):
+            print("# A large number of 'B' quality scores (value 2, ASCII 66) "
+                  "were detected, which makes it likely that this encoding is "
+                  "Illumina-1.5, which has been returned as the only option.",
+                  file=sys.stderr)
+            valid = ['Illumina-1.5']
```
Comment (brentp) [inline reads-utils/guess-encoding.py]: I would also check that `Illumina-1.5` is in the current `valid` list before doing this.

**Review by brentp — CHANGES_REQUESTED** [review]

File: reads-utils/guess-encoding.py (hunk 1 above)
Comment (cviner) [reply reads-utils/guess-encoding.py]: Thanks—will do!

**Comment by cviner** [comment]
@brentp Thanks for the detailed code review and for those improvements!

I'll do that [outstanding one](https://github.com/brentp/bio-playground/pull/23#discussion_r236104810) and run a quick test. I hope to get to it sometime this week and will let you know once it's done.

File: reads-utils/guess-encoding.py (hunk 1 above)
Comment (cviner) [reply reads-utils/guess-encoding.py]: This is already done, I believe sufficiently, at the above block.

Comment (brentp) [reply reads-utils/guess-encoding.py]: indeed.

**Comment by cviner** [comment]
@brentp I've revised the code accordingly, run a couple of quick tests, and pushed the revisions. Thanks again for all your help!

**Review by brentp — APPROVED** [review]

**Comment by brentp** [comment]
thanks for the improvements!

**Pull request merged**